cl/corpus_importer/signals.py

```diff
@@ -0,0 +1,126 @@
+    update_lock_key = make_update_pacer_case_id_key(court_id)
+    # ttl one hour.
+    lock_value = acquire_redis_lock(r, update_lock_key, 60 * 1000)
+
+    highest_known_pacer_case_id = int(
+        r.hget("iquery:highest_known_pacer_case_id", court_id) or 0
+    )
+    iquery_pacer_case_id_current = int(
+        r.hget("iquery:pacer_case_id_current", court_id) or 0
+    )
+    incoming_pacer_case_id = int(docket.pacer_case_id)
+    found_higher_case_id = False
+    if incoming_pacer_case_id > highest_known_pacer_case_id:
+        r.hset(
+            "iquery:highest_known_pacer_case_id",
+            court_id,
+            incoming_pacer_case_id,
+        )
+        found_higher_case_id = True
+
+    if found_higher_case_id:
+        task_scheduled_countdown = 0
+
+        while iquery_pacer_case_id_current + 1 < incoming_pacer_case_id:
+            iquery_pacer_case_id_current += 1
+            task_scheduled_countdown += 1
+            # Schedule the next task with a 1-second countdown increment
+            make_docket_by_iquery_sweep.apply_async(
+                args=(court_id, iquery_pacer_case_id_current),
+                kwargs={"avoid_trigger_signal": True},
+                countdown=task_scheduled_countdown,
+                queue=settings.CELERY_IQUERY_QUEUE,
+            )
+
+        # Update the iquery_pacer_case_id_current in Redis
+        r.hset(
+            "iquery:pacer_case_id_current",
+            court_id,
+            iquery_pacer_case_id_current,
+        )
+
+    # Release the lock once the whole process is complete.
+    release_redis_lock(r, update_lock_key, lock_value)
+
+
+@receiver(
+    post_save,
+    sender=Docket,
+    dispatch_uid="handle_update_latest_case_id_and_schedule_iquery_sweep",
+)
+def handle_update_latest_case_id_and_schedule_iquery_sweep(
+    sender, instance: Docket, created=False, update_fields=None, **kwargs
+) -> None:
+    """post_save Docket signal receiver to handle
+    update_latest_case_id_and_schedule_iquery_sweep
+    """
+
+    if not settings.IQUERY_SWEEP_UPLOADS_SIGNAL_ENABLED and getattr(
+        instance, "avoid_trigger_signal", True
+    ):
+        # If the signal is disabled for uploads in general and the instance
+        # doesn't have avoid_trigger_signal set, abort it. This is a Docket
+        # created by an upload or another RECAP source different from the
+        # iquery probe daemon.
+        return None
+
+    if getattr(instance, "avoid_trigger_signal", False):
+        # This is an instance added by the probe_iquery_pages task
+        # or the iquery sweep scraper that should be ignored (no the highest
+        # pacer_case_id)
+        return None
+
+    # Only call update_latest_case_id_and_schedule_iquery_sweep if this is a
+    # new RECAP district or bankruptcy docket with pacer_case_id not added by
+    # iquery sweep tasks.
+    if (
+        created
+        and instance.pacer_case_id
+        and getattr(instance, "court", None)
```

Comment:
Is this check required? do we have any Docket records that haven't been linked to a specific court?

Reply:
Yeah, that check is not required. Court is a required relation. Also confirmed that in the dev DB there's no Docket with a court Null. Removed it.